### PR TITLE
INT-3804: WatchServiceDirScan: add OVERFLOW event

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-3804

### CODE REVIEW

File: spring-integration-file/src/test/java/org/springframework/integration/file/WatchServiceDirectoryScannerTests.java [hunk 1]
```diff
@@ -86,6 +87,34 @@ public void testInitialAndAddMore() throws Exception {
...
+		*/
+		List<File> filesForOverflow = new ArrayList<File>(512);
+
+		for (int i = 0; i < 512; i++) {
+			filesForOverflow.add(this.folder.newFile("" + i));
+		}
+
+		files = scanner.listFiles(folder.getRoot());
+
+		assertEquals(512, files.size());
```
Comment (garyrussell) [inline spring-integration-file/src/test/java/org/springframework/integration/file/WatchServiceDirectoryScannerTests.java]: This does not pass on OS X - investigating...

File: spring-integration-file/src/test/java/org/springframework/integration/file/WatchServiceDirectoryScannerTests.java [hunk 2]
```diff
@@ -86,6 +87,34 @@ public void testInitialAndAddMore() throws Exception {
...
+
+		for (File fileForOverFlow : filesForOverflow) {
+			files.contains(fileForOverFlow);
+		}
```
Comment (garyrussell) [inline spring-integration-file/src/test/java/org/springframework/integration/file/WatchServiceDirectoryScannerTests.java]: This is an O(n2) search - create a `Set` from `files`.

**Comment by garyrussell** [comment]
Polish here: https://github.com/garyrussell/spring-integration/commit/1e38fb5cf383c3f03dd9423e3bf9b565b9eceaa9

**Comment by artembilan** [comment]
Gary, looks like Travis has overcame the `file` module already, so we can merge the PR.

**Comment by garyrussell** [comment]
Merged as f607f83bf2d3bac92e26ca0339a5bb1d3eae50f1

**Pull request closed without merging**